src/polus/plugins/_plugins/classes/plugin_classes.py

```diff
@@ -97,18 +101,34 @@ def refresh():
             if file.suffix == ".py":
                 continue
 
-            plugin = validate_manifest(file)
-            key = name_cleaner(plugin.name)
-            # Add version and path to VERSIONS
-            if key not in PLUGINS:
-                PLUGINS[key] = {}
-            if plugin.version in PLUGINS[key]:
-                if not file == PLUGINS[key][plugin.version]:
-                    raise DuplicateVersionFound(
-                        "Found duplicate version of plugin %s in %s"
-                        % (plugin.name, PLUGIN_DIR)
-                    )
-            PLUGINS[key][plugin.version] = file
+            try:
+                plugin = validate_manifest(file)
+            except InvalidManifest as e:
+                _invalid[_load_manifest(file)["name"]] = str(e.__cause__)
+            else:
+                key = name_cleaner(plugin.name)
+                # Add version and path to VERSIONS
+                if key not in PLUGINS:
+                    PLUGINS[key] = {}
+                if plugin.version in PLUGINS[key]:
+                    if not file == PLUGINS[key][plugin.version]:
+                        raise DuplicateVersionFound(
+                            "Found duplicate version of plugin %s in %s"
+                            % (plugin.name, PLUGIN_DIR)
+                        )
+                PLUGINS[key][plugin.version] = file
+
+    if len(_invalid) > 0:
+        logger.warning(
+            f"local manifests {[str(x) for x in _invalid.keys()]} are invalid. Run polus.plugins.print_invalid() for more details."
+        )
+
+
+def print_invalid():
+    """Print invalid manifests with respective validation errors."""
+    for x, y in _invalid.items():
+        print(x)
```

Comment:
My preference will be to do away with the global var and just print the error messages. Maybe a summary, and then the detail fails. In my opinion print_invalid() should not be part of the api. 
If we really wanted to have validation has part of the api, I would suggest adding a validate_plugin_manifests() or something along those lines.

Reply:
Removed this method